simulation/model.py

```diff
@@ -42,16 +45,16 @@ def __init__(
         self.household_num_lookahead_years = household_num_lookahead_years
         self.heating_system_hassle_factor = heating_system_hassle_factor
         self.interventions = interventions or []
-        self.air_source_heat_pump_discount_factor_2022 = (
-            air_source_heat_pump_discount_factor_2022
-        )
         self.boiler_upgrade_scheme_cumulative_spend_gbp = 0
         self.gas_oil_boiler_ban_datetime = gas_oil_boiler_ban_datetime
         self.fuel_price_gbp_per_kwh = {
             HeatingFuel.GAS: price_gbp_per_kwh_gas,
             HeatingFuel.ELECTRICITY: price_gbp_per_kwh_electricity,
             HeatingFuel.OIL: price_gbp_per_kwh_oil,
         }
+        self.air_source_heat_pump_price_discount_schedule = sorted(
+            air_source_heat_pump_price_discount_schedule
+        )
```

Comment:
If we have an optional input, then we can change this to:
```suggestion
        ) if air_source_heat_pump_discount_schedule else None
```
I dont even think we need to instantiate an empty list here after the `else` because you check `if self.air_source_heat_pump_discount_schedule` in the `air_source_heat_pump_discount_factor` property.